packages/e2e-tests/specs/experiments/navigation-editor.test.js

```diff
@@ -432,109 +400,57 @@ describe( 'Navigation editor', () => {
-				} ),
-				...getMenuItemMocks( { GET: [] } ),
-			] );
-
+		it( 'saves menu name changes', async () => {
+			await createMenu( { name: initialMenuName } );
 			await visitNavigationEditor();
 
-			// Wait for the navigation setting sidebar.
-			await page.waitForSelector( '.edit-navigation-sidebar' );
-		} );
-
-		afterEach( async () => {
-			await setUpResponseMocking( [] );
-		} );
-
-		it( 'is displayed in inspector additions', async () => {
-			const nameControl = await page.$( nameEditorSelector );
-			expect( nameControl ).toBeDefined();
-		} );
-
-		it( 'saves menu name upon clicking save button', async () => {
-			const newName = 'newName';
-			const menuPostResponse = {
-				id: 4,
-				description: '',
-				name: newName,
-				slug: 'main-menu',
-				meta: [],
-				auto_add: false,
-			};
-
-			await setUpResponseMocking( [
-				...getMenuMocks( {
-					GET: [ menuPostResponse ],
-					POST: menuPostResponse,
-				} ),
-				...getMenuItemMocks( { GET: [] } ),
-			] );
-
-			// Ensure there is focus.
-			await page.focus( inputSelector );
+			// Rename the menu and save it.
+			const newName = 'New menu';
+			await page.waitForSelector( inputSelector );
+			await page.click( inputSelector );
 			await pressKeyTimes( 'Backspace', initialMenuName.length );
 			await page.keyboard.type( newName );
-
-			const saveButton = await page.$(
-				'.edit-navigation-toolbar__save-button'
-			);
-			await saveButton.click();
+			await page.click( '.edit-navigation-toolbar__save-button' );
 			await page.waitForSelector( '.components-snackbar' );
+			await page.reload();
+
+			// Expect the header to have the new name.
 			const headerSubtitle = await page.waitForSelector(
 				'.edit-navigation-menu-actions__subtitle'
 			);
-			expect( headerSubtitle ).toBeTruthy();
 			const headerSubtitleText = await headerSubtitle.evaluate(
 				( element ) => element.innerText
 			);
 			expect( headerSubtitleText ).toBe( newName );
 		} );
 
 		it( 'does not save a menu name upon clicking save button when name is empty', async () => {
-			const menuPostResponse = {
-				id: 4,
-				description: '',
-				name: initialMenuName,
-				slug: 'main-menu',
-				meta: [],
-				auto_add: false,
-			};
-
-			await setUpResponseMocking( [
-				...getMenuMocks( {
-					GET: [ menuPostResponse ],
-					POST: menuPostResponse,
-				} ),
-				...getMenuItemMocks( { GET: [] } ),
-			] );
+			await createMenu( { name: initialMenuName } );
+			await visitNavigationEditor();
 
-			// Ensure there is focus.
-			await page.focus( inputSelector );
+			// Try saving a menu with an empty name.
+			await page.waitForSelector( inputSelector );
+			await page.click( inputSelector );
 			await pressKeyTimes( 'Backspace', initialMenuName.length );
-
-			const saveButton = await page.$(
-				'.edit-navigation-toolbar__save-button'
+			await page.click( '.edit-navigation-toolbar__save-button' );
+			const snackbar = await page.waitForSelector(
+				'.components-snackbar'
```

Comment:
This is the selector that causes failure locally for me.

How reliable is this method for checking if the Menu is saved? I remember Widgets had flaky tests because of the Snackbars, but this might not be related.

Reply:
Yep, unusual, I'm not sure why, it passes with the `--puppeteer-interactive` flag.

Anyhow, I changed it to this
```js
const snackbar = await page.waitForSelector(
    '.components-snackbar',
    { visible: true }
);
```

and it makes it work again. 

I'm aware of some other bugs in `puppeteer` related to `waitForSelector`, and using `visible: true` force puppeteer to use a different waiting strategy (`MutationObserver` is the default, `visible` changes the strategy to polling), so it must be something related to that.

Reply:
> and using visible: true forces puppeteer to use a different waiting strategy

I wasn't aware of this. Thanks for the information.